cvmfs/s3fanout.h

```diff
@@ -236,24 +206,33 @@ class S3FanoutManager : SingleCopy {
   static const unsigned kThrottleReportIntervalSec;
   static const unsigned kDefaultHTTPPort;
 
+  struct S3Config {
+    std::string access_key;
+    std::string secret_key;
+    std::string hostname_port;
+    AuthzMethods authz_method;
+    std::string region;
+    std::string bucket;
+    bool dns_buckets;
+    uint32_t pool_max_handles;
+    unsigned opt_timeout_sec = 20;
```

Comment:
I think this syntax does not work with older C++ compilers, but there has to be an explicit default constructor.